### PR TITLE
chart errors: add test for ensuring we can marshal/unmarshal json

### PR DESCRIPTION
**Pull request by mikeykhalil**
Here is a go playground to shows a simple case of the errors we were running into when unmarshaling:
https://play.golang.org/p/Ued9Xy0TBPZ

The proposed fix is to always omit the error type in the Chart Error. While we lose a bit of information here, it's hard to really make any guarantees about the contents of the error since it's an interface type. 

We could potentially create our own MarshalJSON() / UnmarshalJSON() methods too if we prefer. However, this fix is a bit simpler and would still encode/decode the relevant information users would want to see (what pods failed and why).

### CODE REVIEW

**Review by wwalexander — APPROVED** [review]
Looks good to me. However, looking at the references in the code to HelmError, it looks like it's only ever read to call its Error() method ([here](https://github.com/dollarshaveclub/metahelm/blob/85649e9ad70cfe84bd60b5a44e5524de15404124/pkg/metahelm/error.go#L57)), which returns a string anyway. It might be just as simple to make ChartError.HelmError a string and store the results of the Error() method.

**Comment by mikeykhalil** [comment]
> Looks good to me. However, looking at the references in the code to the HelmError field, it looks like it's only ever read to call its Error() method ([here](https://github.com/dollarshaveclub/metahelm/blob/85649e9ad70cfe84bd60b5a44e5524de15404124/pkg/metahelm/error.go#L57)), which returns a string anyway. It might be just as simple to make ChartError.HelmError a string and store the results of the Error() method, and allow us to keep the error information.

Good point. If we are comfortable changing the type of this field and accept that downstream applications might need to make some updates when bumping versions of metahelm, then we can go with this approach. 

If we wanted to be extra cautious, we could just create a new Field in the ChartError struct (e.g. `HelmErrorMessage`) to store the string, and then still omit the original HelmError. 

Something like 

```
type ChartError struct {
   HelmError error `json:"-"`
   HelmErrorMessage string `json:"helm_error_message"`
   ... 
````

Then consider deprecating the `ChartError.HelmError` and plan to remove it in a later version of metahelm.

File: pkg/metahelm/error.go [hunk 1]
```diff
@@ -30,7 +30,10 @@ type FailedPod struct {
 // be in a failed state, including up to MaxPodLogLines of log data for each.
 type ChartError struct {
 	// HelmError is the original error returned by Helm
-	HelmError error `json:"helm_error"`
+	// We always omit this value when json marshaling/unmarshaling since we would otherwise have to implement the UnmarshalJSON and MarshalJSON methods.
+	HelmError error `json:"-"`
+	// HelmErrorString is the error string of the original error returned by Helm
+	HelmErrorString string `json:"error"`
```
Comment (bkeroackdsc) [inline pkg/metahelm/error.go:36]: The JSON tag should be more descriptive. `helm_error_string` for example.

File: pkg/metahelm/error.go [hunk 2]
```diff
@@ -45,6 +48,7 @@ type ChartError struct {
 func NewChartError(err error) ChartError {
 	return ChartError{
 		HelmError:         err,
+		HelmErrorString:   err.Error(),
```
Comment (bkeroackdsc) [inline pkg/metahelm/error.go:51]: Check `err` for nil before calling any methods on it. If it is nil, this field should be an empty string.

**Review by bkeroackdsc — CHANGES_REQUESTED** [review]

**Review by bkeroackdsc — APPROVED** [review]

**Pull request merged**